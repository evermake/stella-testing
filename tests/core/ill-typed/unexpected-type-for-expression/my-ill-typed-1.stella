language core;

fn some_func(f : Bool) -> (fn(Bool) -> Nat) {
    return if f
        then
            fn (p : Bool) {
                return succ(0);
            }
        else
            fn (p : Bool) {
                return true;
            };
}

fn main(f : Nat) -> Nat {
  return some_func(true)(true);
}