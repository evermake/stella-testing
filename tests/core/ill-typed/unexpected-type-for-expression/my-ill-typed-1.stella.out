An error occurred during typechecking!
===================
Primary type error:
===================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  Nat
but got
  Bool
when typechecking the expression
  true
========================
Alternative type errors:
========================
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  fn (Bool) -> Nat
but got
  fn (Bool) -> Bool
when typechecking the expression
  fn (p : Bool)
{
  return true
}